<!DOCTYPE html>
<html lang="en">
<head>
<title>Gibson assembly primer design tool</title>
<meta content="Primer design and fragment assembly using Gibson Assembly" name="description" />
<!--#include virtual="header.shtml" --> 
<link rel="stylesheet" href="../css/soft2.css" type="text/css" />
<script src="../js/file-menu.js" type="text/javascript"></script>
<script src="../js/readseq.js" type="text/javascript"></script>
<script src="../js/assembly.js" type="text/javascript"></script>
<script src="../js/DNAtm.js" type="text/javascript"></script>
<script src="../js/dna.js" type="text/javascript"></script>
<script src="../js/dimers.js" type="text/javascript"></script>
<script src="../js/displayseq.js" type="text/javascript"></script>

<script>
function openTab(evt, tabName) {
var i, tabcontent, tablinks;
tabcontent = document.getElementsByClassName("tab");
for (i = 0; i < tabcontent.length; i++) {
tabcontent[i].classList.remove("active");
}
tablinks = document.getElementsByClassName("tablink");
for (i = 0; i < tablinks.length; i++) {
tablinks[i].classList.remove("active");
}
document.getElementById(tabName).classList.add("active");
evt.currentTarget.classList.add("active");
}
</script>
<script>
function DisplaySeqs() {
const number_of_sequences = document.getElementById('number_of_sequences');
let result = Display(document.getElementById('inputText').value);
number_of_sequences.innerHTML = result;
}
</script>
<script>
function analyzeAndDisplay() {
const button = document.getElementById("run_button");
const originalColor = button.style.backgroundColor;
button.style.backgroundColor = "#f31221";
button.textContent = "Running";
button.disabled = true;
setTimeout(() => {
try {
 
	const word = analysis(); 
document.getElementById('analysisResult1').value = word[0];
document.getElementById('analysisResult2').value = word[1];
} catch (err) {
    console.error("Error in analyzeAndDisplay:", err);
}
button.style.backgroundColor = originalColor;
button.textContent = "Generate";
button.disabled = false;
}, 10);
}
</script>

 
</head>
<body id="contact" class="billboard">
<div id="page">
<!--#include virtual="menu.shtml" -->
<div class="span-13 home-coloum1"><br />
<div><a href=".">Back to Web Tools</a></div><br />
<h3>Gibson assembly primer design tool</h3>
<p><a href="https://www.snapgene.com/guides/gibson-assembly" target="_blank">Gibson assembly</a> is a powerful DNA cloning technique that enables the seamless joining of multiple linear DNA fragments into a single construct. If the fragments contain the necessary overlapping regions, they can be assembled into an intact plasmid. This isothermal, single-reaction method relies on the coordinated activity of a 5′ exonuclease, a DNA polymerase, and a DNA ligase to efficiently fuse overlapping DNA sequences. This approach allows for the precise construction of synthetic and natural genes, genetic pathways, and even entire genomes, making it a valuable tool in molecular engineering. For successful assembly, DNA fragments must typically share at least 20 bp of homology with adjacent segments, ensuring a melting temperature (Tm) of ≥50°C in the overlapping region.</p>
<p><strong>Application Principle:</strong> Arrange the DNA fragments in the desired order for assembly. While not mandatory, including the vector sequence at both the beginning and end of the list can help streamline the process.</p><br />
<input type="file" id="fileInput">  
<strong><span class="blockHeader" id="number_of_sequences"></span></strong><br />
<div id="section1" class="section"><strong>Input Format:</strong> Sequences can be pasted or uploaded as a file in FASTA format.<br />
<div class="sub-tab-buttons"><button class="sub-tab-btn active" data-subtab="tab1-1">Target sequences</button>
<button class="sub-tab-btn" data-subtab="tab1-2">Pre-designed list of primers</button></div>
<div id="tab1-1" class="sub-tab-content" style="display: block;">
<textarea id="inputText" style="width:600px;height:250px" oninput="DisplaySeqs()">
>pUC19_left
cggctcgtatgttgtgtggaattgtgagcggataacaatttcacacaggaaacagctatgaccatgattacgccaagcttgcatgcctgcaggtcgactctagag 
>1  
gaagcgctctgtccactacaccatgattacttgggggatgaatttaacaatattcctacgcaagatcgtctagcgcttgatacgtgccgattttatctgtagcaatgatgaaaattaactttctacctaggcagtcacagttttttattttccctaatatagcttcgtccttcatcggttgacaaggcgaaagccacctcgcccgcgctgatgaagttgaagactcgccttgacgttaactagatgccggcgacacccacgctagatgcttgcagtggcactgagttctgctcccgcaaagacact
>2 
cgaagggttcgttaccggcttcaacgatcaacgattcaaccccaagttccaacgtcgaaggtggaaatttcggcggagatggctactcctattctaggacgcgaaatcccggcattgggtaacatttgcccagacgcgccacgggtgcggatatcccgactaggtaacgggcaaggagaaagctggtaggcgcatgatgcaggaattgcagcatcagcctaggtgccgagaacatatcggacgatcttcatcccttctgatagggctaaccctgatctatattcaactgcctttattagtattgtggtcaatttattcataccccgacgcatgttcggatttgacctggtcgagaaccgccgcatacgtgtaatagcagtgtcgat
>3  
cccttgatttggagagcacgcgactagaggtgcctgggaagccctaatagcccgtaataacccttagcggcctgcggtccccccggggagacttatcatgtgtaatcccctcccatcagtgcgcgtgatattttatgcagagtaaaccatgaaagataaacgacactctaacgatcgtataatgttcaccccgcgtgcgatattgtcccgacccacattgcactttgcgtcacttggccccgatacctaaactatgctgaggga
>4 
tacgggtaatgtcttactcacgctctcgaaagaacgtgtgtggtagaaaggtaatacattacctctgtgctagttacgacgcgccggtgccttctctcggagacgctcagatggtgtagggtatggggtgtaccatcgttttggattgcgaagactaaggctgcatttagttcaaccgtcctgccggtattacccga
>pUC19_right
gatccccgggtaccgagctcgaattcactggccgtcgttttacaacgtcgtgactggg
</textarea>
</div>
<div id="tab1-2" class="sub-tab-content"><textarea id="inputPrimerList" style="width:600px;height:300px" placeholder=">M13-47&#13;cgccagggttttcccagtcacgac&#13;>RP&#13;tttcacacaggaaacagctatgac&#13;>RP-48&#13;agcggataacaatttcacacagga&#13;&#10;"></textarea></div>
</div><br />

<table border="0">
<tr><td colspan="2" style="text-align: left;"><span><strong>Primer design option</strong></span></td></tr>
<tr><td>Length range (nt):</td><td><input style="text-align:center;" type="number" id="minlen" min="12" max="99" value="18" />-<input style="text-align:center;" type="number" id="maxlen" min="12" max="99" value="23"/></td></tr>
<tr><td>Tm range (°C):</td><td><input style="text-align:center;" type="number" id="mintm" min="37" max="80" value="60"/>-<input style="text-align:center;" type="number" id="maxtm" min="37" max="90" value="62"/></td></tr>
<tr><td>Minimal linguistic complexity (%):</td><td><input style="text-align:center;" type="number" id="minlc" min="10" max="90" value="75"/></td></tr></tr>
<tr><td>Variants of the 3'-end composition (5'-3'):</td><td><input style="text-align:left;" type="text" name="end3com" id="end3com" value="n"/></td></tr>

<tr><td colspan="2" style="text-align: left;"></td></tr>
<tr><td><button onclick="analyzeAndDisplay()" id="run_button" class="button button2">Generate</button></td><td><button onclick="document.getElementById('inputText').value= '';document.getElementById('analysisResult1').value = ''; document.getElementById('analysisResult2').value = '';" class="button button1">Clear</button></td></tr>
</table>

<span><strong>To export the results: select all (Ctrl-A), copy (Ctrl-C) and paste (Ctrl-V) to Excel sheet.</strong></span><br />
<div id="section2" class="section">
<div class="sub-tab-buttons"><button class="sub-tab-btn active" data-subtab="tab2-1">Primer list</button><button class="sub-tab-btn" data-subtab="tab2-2">PCR primer pairs</button></div>
<div id="tab2-1" class="sub-tab-content" style="display: block;"><textarea id="analysisResult1" style="width:600px;height:300px"></textarea></div>
<div id="tab2-2" class="sub-tab-content"><textarea id="analysisResult2" style="width:600px;height:300px"></textarea></div>
</div><br /><br />


<li>Sequences are expected to be represented in the standard <a href="https://blast.ncbi.nlm.nih.gov/doc/blast-topics/" target="_blank">IUB/IUPAC nucleic acid codes</a> are acceptable letters: <b>B</b>=(C,G,T), <b>D</b>=(A,G,T), <b>H</b>=(A,C,T), <b>K</b>=(G,T), <b>M</b>=(A,C), <b> N</b>=(A,C,G,T), <b>R</b>=(A,G), <b>S</b>=(G,C), <b>V</b>=(A,C,G), <b>W</b>=(A,T), <b>Y</b>=(C,T), <b>U</b>=Uracil, <b>I</b>=Inosine.</li> 
<li><strong>Variants of the 3'-end composition (5'-3')</strong>: the structure of the last nucleotides at the 3'-end of the primer, it can be specified by "<span class="blue"><strong>N</strong></span>" for any pattern, or it can be encoded by one, two, three or more characters of standard or mixed letters. It is possible to specify one or more patterns (separated by spaces and of <strong>equal length</strong>): <strong><span class="blue">sws ssw sww wss www</strong></span>. For example, the pattern <span class="blue"><strong>WSS</strong></span> corresponds to all variants of the 3'-end composition: <span class="blue">acc acg agc agg tcc tcg tgc tgg</span>.</li>
<li><strong>Pre-designed list of primers/probes</strong>. Pre-designed primers/probes list is used for multiplexing with prior designed PCR primer/probe sets.</li> 
<li>Linguistic sequence complexity (LC%) is a measure of the 'vocabulary richness' of a genetic text based on counting the number of possible combinations of nucleotides ("entropy" of the set of possibilities) to the theoretically possible one. This value for sequence is converted to percentages, 100% being the highest level.</li> 
 
<br /><br />
</div>

 
<script src="../js/tabscript.js"></script>
<!--#include virtual="copyright.shtml" --> 
<!--#include virtual="footer.shtml" --> 
</body>
</html>