<!DOCTYPE html>
<html lang="en">
<head>
<title>KASP primers assay design tool</title>
<meta content="KASP primers assay design tool" name="description" />
<!--#include virtual="header.shtml" --> 
<link rel="stylesheet" href="../css/soft2.css" type="text/css" />
<script src="../js/file-menu.js" type="text/javascript"></script>
<script src="../js/readseq.js" type="text/javascript"></script>
<script src="../js/kasp2.js" type="text/javascript"></script>
<script src="../js/melting.js" type="text/javascript"></script>
<script src="../js/dna.js" type="text/javascript"></script>
<script src="../js/dimers.js" type="text/javascript"></script>
<script src="../js/displayseq.js" type="text/javascript"></script>
<script src="../js/ensembl.js" type="text/javascript"></script>
<script>
function openTab(evt, tabName) {
var i, tabcontent, tablinks;
tabcontent = document.getElementsByClassName("tab");
for (i = 0; i < tabcontent.length; i++) {
tabcontent[i].classList.remove("active");
}
tablinks = document.getElementsByClassName("tablink");
for (i = 0; i < tablinks.length; i++) {
tablinks[i].classList.remove("active");
}
document.getElementById(tabName).classList.add("active");
evt.currentTarget.classList.add("active");
}
</script>
<script>
function analyzeAndDisplay() {   
var word = analysis();
document.getElementById('analysisResult1').value = word[0];
document.getElementById('analysisResult2').value = word[1];
}
</script>
<script>
function DisplaySeqs() {
const number_of_sequences = document.getElementById('number_of_sequences');
let result = Display(document.getElementById('inputText').value);
number_of_sequences.innerHTML = result;
}
</script>
<script>
function handleClickTaqMan(checkbox){
const taqmancheckbox = document.getElementById('taqmanprobe');
if (taqmancheckbox.checked) {
const checkbox = document.getElementById('mgbprobe');
checkbox.checked = false;}
}
</script>
<script>
function handleClickMGB(checkbox){
const mgbcheckbox = document.getElementById('mgbprobe');
if (mgbcheckbox.checked) {
const checkbox = document.getElementById('taqmanprobe');
checkbox.checked = false;}  
}
</script>
</head>
<body id="contact" class="billboard">
<div id="page">
<!--#include virtual="menu.shtml" -->
<div class="span-13 home-coloum1"><br />
<div><a href=".">Back to Web Tools</a></div><br />
 
<h3>KASP primers genotyping assay design tool</h3>
<p>Allele-Specific PCR (AS-PCR) - <a href="https://www.biosearchtech.com/support/education/kasp-genotyping-reagents/kasp-overview" target="_blank">KASP</a>™ (Kompetitive Allele Specific PCR) or <a href="https://eu.idtdna.com/pages/products/qpcr-and-pcr/genotyping/pace-snp-genotyping-assays" target="_blank">PACE</a>™ (PCR Allele Competitive Extension) or Allele-Specific Quantitative PCR (<a href="https://www.frontiersin.org/journals/molecular-biosciences/articles/10.3389/fmolb.2022.773956/full" target="_blank">ASQ</a>) - based genotyping assay designs for multiallelic discrimination of single nucleotide polymorphisms (<b>SNPs</b>) and insertions and deletions (<b>InDels</b>) at specific loci. The application provides professional facilities for genotyping assay design for SNP/InDel-specific KASP assay-targeting primers (KASP Assay Mix).</p>

<p><strong>Input format:</strong> Sequence(s) can be pasted or uploaded as a file in FASTA format or retrieved flanked sequence SNP sequences from the Ensembl database. One or multiple SNP/variant <strong>rsID</strong>s (comma or space separated: rs4988235 rs1357617 rs2046361 rs717302 rs917118) were used to retrieve the surrounding sequence (±flank bases) and enter the species name (<code>homo_sapiens</code>, <code>mus_musculus</code>, <code>rattus_norvegicus</code>, <code>bos_taurus</code>, <code>danio_rerio</code>, <code>arabidopsis_thaliana</code> etc).<br /><strong>Size Limitations:</strong> The length of the query sequence and size of the batch file are theoretically unlimited.</p>
<p>Alternatively, direct retrieval of flanked SNP sequences in the <a href="kasp2.html">human genome</a>.</p>

<div>
<label for="speciesInput">Species:</label><input type="text" id="speciesInput" value="homo_sapiens" size="15" value="homo_sapiens" />
<label for="variantInput">rsIDs:</label><input type="text" id="variantInput" value="rs4988235 rs1357617 rs2046361 rs717302 rs917118" size="50" /><br />
<label for="flankInput">Flank size (&plusmn; bases):</label><input type="number" size="15" id="flankInput" value="300" min="100" />
<button onclick="fetchVariants()">Retrieve Sequences</button></div><br />

<input type="file" id="fileInput">  
<strong><span class="blockHeader" id="number_of_sequences"></span></strong><br />
<div id="section1" class="section">
<strong>Input Format:</strong> Sequence(s) can be pasted or uploaded as a file in FASTA format.<br />
<div class="sub-tab-buttons">
<button class="sub-tab-btn active" data-subtab="tab1-1">Target sequence(s)</button>
<button class="sub-tab-btn" data-subtab="tab1-2">Allele-specific primer tails list</button>
</div>
<div id="tab1-1" class="sub-tab-content" style="display: block;"><textarea id="inputText" style="width:600px;height:300px" oninput="DisplaySeqs()">
>1&#13;&#10;tgaactacggcgtgcgactaccgcggataaacctgtgtaaagaatataagtgttactcg [R] tcaattccgcgatctagttaatccgcttcgttagtcgtattcatgggacagaaattatggacgct&#13;&#10;
>2&#13;&#10;tgtccatatgtcgaacggtcagagaccgctgacactagtgca [/ATAGACGTCGATCGT] gttgctccacatggcagaaagaaatgaggatgagtggaaaatccataatcacgatgaac&#13;&#10;
>3&#13;&#10;tgggcagcattagtagaagaaagtacaagaccgtgtgtagaggatactct [GATATACTTGAG/CAGTCC] agcagatagcgttggataggcgacaggattattggagcgccgtcgagaac&#13;&#10;
>4&#13;&#10;caaacaccaaactggtgagtcgtggtttacaacacgggagttcaaaactg [TATCCGAATGACGAATGTTCACGTCCTTAAAC/CATCCGAATCACGAATGTTCAGTTCCTTTAAG] catcatgaaatgagtttagtttgggtggctcgtaagtagacataaggcac&#13;&#10;
</textarea></div>
<div id="tab1-2" class="sub-tab-content"><textarea id="inputTails" style="width:600px;height:300px" placeholder=">FAM&#13;GAAGGTGACCAAGTTCATGCT&#13;>HEX&#13;GAAGGTCGGAGTCAACGGATT&#13;"></textarea></div>
</div>
<br />

<table border="0">
<tr><td colspan="3" style="text-align: left;"><span><strong>Reverse primer design option</strong></span></td></tr>
<tr><td>Length range (nt):</td><td><input style="text-align:center;" type="number" id="minlen" min="12" max="99" value="18" />-<input style="text-align:center;" type="number" id="maxlen" min="12" max="99" value="28"/></td><td><input type="checkbox" id="repeats" name="repeats" value="repeats" checked ><label for="repeats">Non-specific priming control</label></td></tr>
<tr><td>Tm range (°C):</td><td><input style="text-align:center;" type="number" id="mintm" min="37" max="80" value="60"/>-<input style="text-align:center;" type="number" id="maxtm" min="37" max="90" value="62"/></td><td><input type="checkbox" id="overlapping" name="overlapping" value="overlapping" ><label for="overlapping">Overlaping primers</label></td></tr>
<tr><td>Minimal Linguistic Complexity (%):</td><td><input style="text-align:center;" type="number" id="minlc" min="10" max="90" value="75"/></td><td><input type="checkbox" id="ctconvert" name="ctconvert" value="ctconvert"><label for="ctconvert">C>>T bisulfite conversion</label></td></tr></tr>
<tr><td>Variants of the 3'-end composition (5'-3'):</td><td colspan="2" style="text-align: left;"><input style="text-align:left;" type="text" name="end3com" id="end3com" size="35" value="sws ssw sww wss www"/></td></tr>
<tr><td colspan="3" style="text-align: left;"></td></tr>
<tr><td>SNP located at the 3'-end of <strong>ASP</strong> (1..):</td><td><input style="text-align:center;" type="number" id="snp" min="1" max="30" value="2"/></td><td></td></tr>
<tr><td colspan="3" style="text-align: left;"></td></tr>
<tr><td colspan="2" style="text-align: left;"><button onclick="analyzeAndDisplay()" id="run_button" class="button button2">Generate</button></td>
<td><button onclick="document.getElementById('inputText').value= '';document.getElementById('analysisResult1').value = ''; document.getElementById('analysisResult2').value = '';" class="button button1">Clear</button></td></tr>
</table>

<span><strong>To export the results: select all (Ctrl-A), copy (Ctrl-C) and paste (Ctrl-V) to Excel sheet:</strong></span><br /> 
<div id="section2" class="section">
<div class="sub-tab-buttons"><button class="sub-tab-btn active" data-subtab="tab2-1">Primer list</button><button class="sub-tab-btn" data-subtab="tab2-2">KASP assay sets</button></div>
<div id="tab2-1" class="sub-tab-content" style="display: block;"><textarea id="analysisResult1" style="width:600px;height:300px"></textarea></div>
<div id="tab2-2" class="sub-tab-content"><textarea id="analysisResult2" style="width:600px;height:300px"></textarea></div>
</div>

<br /> 
<ul>
<li>Sequences are expected to be represented in the standard <a href="https://blast.ncbi.nlm.nih.gov/doc/blast-topics/" target="_blank">IUB/IUPAC nucleic acid codes</a> are acceptable letters: <b>B</b>=(C,G,T), <b>D</b>=(A,G,T), <b>H</b>=(A,C,T), <b>K</b>=(G,T), <b>M</b>=(A,C), <b> N</b>=(A,C,G,T), <b>R</b>=(A,G), <b>S</b>=(G,C), <b>V</b>=(A,C,G), <b>W</b>=(A,T), <b>Y</b>=(C,T), <b>U</b>=Uracil, <b>I</b>=Inosine.</li> 
<li><strong>Variants of the 3'-end composition (5'-3')</strong>: the structure of the last nucleotides at the 3'-end of the primer, it can be specified by "<span class="blue"><strong>N</strong></span>" for any pattern, or it can be encoded by one, two, three or more characters of standard or mixed letters. It is possible to specify one or more patterns (separated by spaces and of <strong>equal length</strong>): <strong><span class="blue">sws ssw sww wss www</strong></span>. For example, the pattern <span class="blue"><strong>WSS</strong></span> corresponds to all variants of the 3'-end composition: <span class="blue">acc acg agc agg tcc tcg tgc tgg</span>.</li>
<li>Linguistic sequence complexity (LC%) is a measure of the 'vocabulary richness' of a genetic text based on counting the number of possible combinations of nucleotides ("entropy" of the set of possibilities) to the theoretically possible one. This value for sequence is converted to percentages, 100% being the highest level. For some <strong>Allele-Specific Primers (ASP)</strong> there will be a value of <b><span class="red">LC%=0</span></b>, indicating the presence of a dimer for this primer, and attention should be paid to this fact.</li> 
<li>The input contains nucleotide sequences of allelic variants which are used to compute primers for Kompetitive Allele Specific PCR (<a href="https://www.biosearchtech.com/support/education/kasp-genotyping-reagents/kasp-overview" target="_blank">KASP</a>) or Allele-Specific Quantitative PCR (<a href="https://www.frontiersin.org/journals/molecular-biosciences/articles/10.3389/fmolb.2022.773956/full" target="_blank">ASQ</a>). In this example, allele-specific PCR is being designed to genotype <b>SNP</b> alleles. The <b>SNP</b> of interest must be enclosed in [square brackets] and can be formatted as [First allele/Second allele/Third allele/Fourth allele] or [<a href="https://blast.ncbi.nlm.nih.gov/doc/blast-topics/" target="_blank">IUPAC code</a>] or [Target Nucleotide].</li> 
<li>Optionally, use two ‘<b><span class="red">/.../</span></b>’ signs for the start and end of the excluded region (this is possible multiple times).</li> 
<li><strong>Primer tails list</strong>: ASP can be designed according standard KASP guidelines (LGC Biosearch Technologies) carrying the standard FAM (5′-GAAGGTGACCAAGTTCATGCT-3′) and HEX (5′-GAAGGTCGGAGTCAACGGATT-3′) tails. Automatically adding 5′-tails to each primer. The input data should have the following FASTA style, although the name should be optional:<br />>FAM <br />GAAGGTGACCAAGTTCATGCT<br />>HEX <br />GAAGGTCGGAGTCAACGGATT</li> 
</ul><br />

<p><strong>Example 1</strong>. Formatting sequences for <b>SNP</b>:<br />
<tt>>1<br />
gctctctgtgtctgatccaagaggcgaggccagtttcatttgagcattaa <b><span class="red">[A/G]</span></b> tgtcaagttctgcacgctatcatcatcaggggccgaggcttctctttgtt<br />
>2<br />
tcatattccagtttgggcgagttttaagataggtccgg <b><span class="red">[S]</span></b> acagtctttgcggcgccaacgcgtctttctccagcagacagtccccggactgc<br />
>3<br />
tcatattccagtttgggcgagttttaagataggtccgg <b><span class="red">[C]</span></b> acagtctttgcggcgccaacgcgtctttctccagcagacagtccccggactgc</tt></p>

<p>Allele-specific PCR (AS-PCR) assays can be designed for discrimination of insertions/deletions (<b>InDels</b>) polymorphisms. The program imposes no size limit on length difference for InDels alleles, provided that all alleles can be aligned, and the alignment has a sufficient length of overlapping to target primers. Two, three or four allelic variants may be included for analysis to compute allele-specific primers (ASPs). Only differences between the variants must be shown within the brackets. One Allele-Specific Primer (ASP) will be computed for each of the input allelic variants. Also, one common primer (Universal Primer, UP) will be computed which targets a conserved region in all sequences. Input sequences should be in a FASTA format with differences between alleles placed within square brackets: [allele1/allele2/allele3/allele4].</p>
<p><strong>Example 2</strong>. Formatting sequences for <b>InDels</b>:<br />
<tt>>1<br />
tcatattccagtttgggcgagttttaagataggtccgg <b><span class="red">[AG/]</span></b> acagtctttgcggcgccaacgcgtctttctccagcagacagtccccggactgc</tt></p>
<p><strong>Example 3</strong>. Detection of <b>Multi-Nucleotide Variants</b> (MNV) is possible using KASP. Sequence information for MNVs should be submitted using the [allele1/allele2] format:<br />
<tt>>1<br /> 
tgggcagcattagtagaagaaagtacaagaccgtgtgtagaggatactct <b>[<span class="red">GATATACTTGAG</span>/<span class="blue">CAGTCC</span>]</b> agcagatagcgttggataggcgacaggattattggagcgccgtcgagaac</tt></p>
<p><strong>Example 4</strong>. KASP assays can be designed to detect <b>Haplotypes</b> of any size. Sequence information for haplotypes should be submitted using the [allele1/allele2] format. The [square brackets] should surround all SNPs that are part of the haplotype. Any nearby SNPs that are not considered to be part of the haplotype should be outside of the square brackets and should be identified using the appropriate IUPAC ambiguity code.<br />
<tt>>1<br />
caaacaccaaactggtgagtcgtggtttacaacacgggagttcaaaactg<br />
<b>[<span class="blue">T</span>ATCCGAAT<span class="blue">G</span>ACGAATGTTCA<span class="blue">CG</span>TCCTT<span class="blue">A</span>AA<span class="blue">C</span><br />/<span class="red">C</span>ATCCGAAT<span class="red">C</span>ACGAATGTTCA<span class="red">GT</span>TCCTT<span class="red">T</span>AA<span class="red">G</span>]</b><br />
catcatgaaatgagtttagtttgggtggctcgtaagtagacataaggcac</tt></p><br /><br />

<h3>&#1057; &gt;&gt; T bisulfite conversion (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3233226/" target="_blank">bisulfite modified genome</a>)</h3>
Sequence, design of specific PCR primers for <i>in silico</i> bisulphite conversion for both strands - only cytosines not followed by guanidine (CpG methylation) will be replaced by thymines:<br />
 <tt><br /> 5&rsquo;aa<span class="blue">CG</span>aagt<span class="blue">CCCCa</span>-3' &nbsp;&nbsp; &nbsp; &nbsp;&nbsp;5&rsquo;aa<span class="red">CG</span>aagt<span class="blue">TTTTa</span>-3'<br />
 &nbsp;&nbsp;||||||||||||| &nbsp; &nbsp; -&gt; &nbsp; &nbsp; &nbsp;||||||:|::::|</tt>&nbsp;<br />
<tt>3&rsquo;tt<span class="blue">GC</span>tt<span class="blue">C</span>aggggt<span class="blue"></span>-5' &nbsp;&nbsp; &nbsp;&nbsp; &nbsp;3&rsquo;tt<span class="red">GC</span>tt<span class="blue">T</span>aggggt</tt>
<br /><br />
<h3>Non-specific priming control</h3><p>Oligonucleotide specificity is one of the most critical factors for good PCR; optimal primers should hybridize only to the target sequence, especially when using complex genomic DNA as a template. Amplification problems can occur when primers anneal to repetitive sequences (retrotransposons, transposons or inverted tandem repeats). Alternative product amplification can also happen when primers are complementary to inverted repeats and produce multiple bands. However, the generation of inverted repeat sequences is exploited in two common generic DNA fingerprinting methods (RAPD).</p><br /><br />
</div>

<script src="../js/tabscript.js"></script>
<!--#include virtual="copyright.shtml" --> 
<!--#include virtual="footer.shtml" --> 
</body>
</html>