<!DOCTYPE html>
<html lang="en">
<head>
<title>Custom multiplex tiling PCR panel design tool</title>
<meta content="Custom multiplex tiling PCR panel design tool" name="description" />
<!--#include virtual="header.shtml" --> 
<link rel="stylesheet" href="../css/soft2.css" type="text/css" />
<script src="../js/file-menu.js" type="text/javascript"></script>
<script src="../js/readseq.js" type="text/javascript"></script>
<script src="../js/panel.js" type="text/javascript"></script>
<script src="../js/DNAtm.js" type="text/javascript"></script>
<script src="../js/dna.js" type="text/javascript"></script>
<script src="../js/dimers.js" type="text/javascript"></script>
<script src="../js/displayseq.js" type="text/javascript"></script>
<script src="../js/ncbifasta.js" type="text/javascript"></script>
<script>
function openTab(evt, tabName) {
var i, tabcontent, tablinks;
tabcontent = document.getElementsByClassName("tab");
for (i = 0; i < tabcontent.length; i++) {
tabcontent[i].classList.remove("active");
}
tablinks = document.getElementsByClassName("tablink");
for (i = 0; i < tablinks.length; i++) {
tablinks[i].classList.remove("active");
}
document.getElementById(tabName).classList.add("active");
evt.currentTarget.classList.add("active");
}
</script>
<script>
function DisplaySeqs() {
const number_of_sequences = document.getElementById('number_of_sequences');
let result = Display(document.getElementById('inputText').value);
number_of_sequences.innerHTML = result;
}
</script>
<script>
function analyzeAndDisplay() {   
var word = analysis();
document.getElementById('analysisResult1').value = word[0];
document.getElementById('analysisResult2').value = word[1];
}
</script>
 
</head>
<body id="contact" class="billboard">
<div id="page">
<!--#include virtual="menu.shtml" -->
<div class="span-13 home-coloum1"><br />
<div><a href=".">Back to Web Tools</a></div><br />
 
<h3>Custom multiplex tiling PCR panel design tool</h3>
<p>Custom multiplex tiling PCR panel design for amplicon sequencing technology, targeting next-generation (NGS) and third-generation (TGS) DNA sequencing technologies. Multiplex amplicon sequencing is a powerful method suitable for targeted sequencing across a wide range of organisms, including humans, plants, animals, and microorganisms - assuming that reference genome sequences are available. This tool enables users to design custom multiplex tiling PCR panels optimized for high-throughput amplicon sequencing to identify genetic variation in natural or mutagenized populations through eco-tilling or multiplex CRISPR screens. It ensures high primer compatibility within each panel and comprehensive target region coverage.
<br /><strong>Tiling Strategy:</strong> The tool will divide the target region into overlapping amplicons, ensuring full coverage. Two complementary primer panels (A and B) are automatically generated. All primer pairs within each panel are designed to be mutually compatible for simultaneous multiplex PCR reactions, minimizing primer-dimer formation and cross-reactivity.
<br /><strong>Input format:</strong> Sequences can be pasted or uploaded as a file in FASTA format or retrieved sequence (NCBI’s accession, e.g. A02710) from NCBI’s "nuccore" nucleotide database. This approach is particularly effective for capturing genomic regions smaller than 1 megabase (Mb) and is compatible with both short-read (Illumina) and long-read (Oxford Nanopore Technologies) sequencing platforms.
<br /><strong>Desired amplicon size range</strong>, which may vary depending on the sequencing platform: For short-read NGS (Illumina), typical amplicons range from 150-400 bp. For long-read TGS, larger amplicons range from 500-3,000 bp or more may be used.
<br /><strong>Output:</strong> Downloadable list of designed primers with detailed information: sequences, genomic coordinates, amplicon sizes, and melting temperatures. Ready-to-use primer mixes for synthesis or experimental validation.</p>
<div>
<label for="ncbiId">NCBI accession ID:</label><input type="text" id="ncbiId" id="ncbiId" value="NC_045512" />
<button onclick="fetchFastaSequence()">Retrieve Sequence</button>
</div><br />

<input type="file" id="fileInput">  
<strong><span class="blockHeader" id="number_of_sequences"></span></strong><br />
<div id="section1" class="section">
<strong>Input Format:</strong> Sequences can be pasted or uploaded as a file in FASTA format.<br />
<div class="sub-tab-buttons">
<button class="sub-tab-btn active" data-subtab="tab1-1">Target sequence(s)</button>
<button class="sub-tab-btn" data-subtab="tab1-2">Pre-designed list of primers/probes</button>
</div>
<div id="tab1-1" class="sub-tab-content" style="display: block;">
<textarea id="inputText" style="width:600px;height:250px" oninput="DisplaySeqs()">
>1 
[aggcagagagaatacggcttcaccgcaatatgaatcaggcgtagcagtacatccgatggaagcctaaagttcacaaaaccacgttagccacgcgtgcccggcacccgtcgtcgcagacaaagccacgtgagtcgaagtcaggctttgcttctaacgtggtatatcaagtcaagttacgcccccgaaattgcactagctaagttttagcgctctttcggattatcaatatgataaactttacactgtgcataatgaaccgccggggaccaaataatgtttaaggtggataggttgacggccacaccctggcggtaacgcgatcccgaatcggcaaggtgccgctcaccaataaaagagccatttgcaattgagtgattaaactagacccctacatcatggtagtcgcacgggttagtttctacacacgaaaggttgatggttccgccatgccctaaagaaattctgcgcaatgggataggggaaagactgatacactgtaaaaagtgtgtccgtatgtactctacttgtgacttcagtaccacaacttagggaagttaccagaaacgtggagtgactgaaccactgcggggtttggctctagtataagtccgcttggacttgggtttgccccgattaaagactgggttgtggaccggttagcatctatttcaccaggacttatggaactgactgtccactagatgatttaataaccatctcgttgaatctaatttatcgccagggacctgtgaatgtgggacaaggtaaaggcactcctccgggcagatcgcgcgtccattcaccaactgactttagggccggtactaagaaccttaactcgcgcttccattctctgcacataagtcatattcggaagtattgaaccaactctcttaagcgaaaccgcagttggaagagtcagtttaagggtttgatcgagagagggggggctctcaatcacatctgttaacagtacgcgtcagcttgccctagacacgttgtccagctgccgagcccccgttcatacagaaaacaagcaggattgtatgccacgctacacgtattcgacgaagccgcttttgcgcagagcgtgctcggctggcaagaattgatatacgcagctctcctaagcggggataccagagctccaacgtacgtagcagagggtgtcctttagtgcttggtctagaga]
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa 
[tatccacccagtgcacagcgtttaggtatttcatcacggctaatctagtttgacccaattcgacggtaacatcgtgccttcctgagtgtactctgcgggcgatagaaatatactgtcaaataacttctagtgatcgggagcacgaacgtgcaggtaaggaacctaacaaaagaataacgtttggcaaggtccgttaccgatttagttactagtatatagttagttcaccaaagagtaggccgtagtagtactctgtagacgcagacgtgtgaacgggttcctaccctcttccgtccggagcttcagtttggagcatttgaggtagaggctcaagcctcaaggcaaggttccgtctgcatcacataaatatccgcgatttacagtcggaatacggctgcgctgggtggtgctgcttgccgtaagccgacccacgttacaatccttaatctcgagcagatatggggtatagaaaggttatggtatggtgccgtcggcggatagatttagccatttacgtaaagcggccgagttcagtggagcgttaggcctcctcatttactcgttagtccgttcaagggatttatagttggttcttggcttggctcaataccttttacgaaccatgtgatacacacgatgcctgacatagtaataggtaatgagggcgcgggagcctgacggcctcttcccggttacggcgcgtgttagcttatctatagcgaactcgcactgctaagaacagaaacattagtccagatgggctccctatccggtaacatgcgctggaatcctatgcgggctttcgctcaactaatcaggccgagttgcccaacgtcgcatgacttcattttctgctcctcttgtctgcgggagctagatcatcagactaaaccacccacagttttgagcacaccttatgctgcttgtccaacacggctcatccaacccgccgtcggataccatcgcgggaacgattgtgcgcatagcgcactaatgagattaggagttgtccagggataccacccacgatcgactggcttgagttctcctccggacttgaatgataaggatcccatgagagatactatcagcctatctgacacatcggtatgagataccaagtatcctgacccaccattgttctagctccggaaagcccgtgatcgttccagttcaacttcagtagtaagctcttatcatgcgcgatagcagtgcgccttgataaatcgtgcttgcaaacacataaccgttgagagtaaaattgaaattgctcatgggcctggactcgctgtctaattcggttcatatatatgtagagaacaggtgactcgtttggggtcatgtcgagccgttaaatcaaagcacctgggccttaccttaactagacaagcctagcg]
</textarea>
</div>
<div id="tab1-2" class="sub-tab-content"><textarea id="inputPrimerList" style="width:600px;height:300px" placeholder=">M13-47&#13;cgccagggttttcccagtcacgac&#13;>RP&#13;tttcacacaggaaacagctatgac&#13;>RP-48&#13;agcggataacaatttcacacagga&#13;&#10;"></textarea></div>
</div><br />

<table border="0">
<tr><td colspan="3" style="text-align: left;"><span><strong>Primer design option</strong></span></td></tr>
<tr><td>Length range (12-100 nt):</td><td colspan="2" style="text-align: left;"><input style="text-align:center;" type="number" id="minlen" min="12" max="99" value="20" />-<input style="text-align:center;" type="number" id="maxlen" min="12" max="99" value="21"/></td></tr>
<tr><td>Tm range (°C):</td><td colspan="2" style="text-align: left;"><input style="text-align:center;" type="number" id="mintm" min="37" max="80" value="62"/>-<input style="text-align:center;" type="number" id="maxtm" min="37" max="90" value="64"/></tr>
<tr><td>Minimal linguistic complexity (70-90%):</td><td colspan="2" style="text-align: left;"><input style="text-align:center;" type="number" id="minlc" min="10" max="90" value="75"/></td></tr>
<tr><td>Variants of the 3'-end composition (5'-3'):</td><td colspan="2" style="text-align: left;"><input style="text-align:left;" type="text" name="end3com" size="45" id="end3com" value="sws ssw sww wss www"/></td></tr>
<tr><td><input type="checkbox" id="repeats" name="repeats" value="repeats" checked ><label for="repeats">Non-specific priming control</label></td><td colspan="2" style="text-align: left;"><input type="checkbox" id="overlapping" name="overlapping" value="overlapping" ><label for="overlapping">Overlaping primers</label></td></tr>
<tr><td></td><td colspan="2" style="text-align: left;"><input type="checkbox" id="ctconvert" name="ctconvert" value="ctconvert"><label for="ctconvert">C>>T bisulfite conversion</label></td></tr>
<tr><td>Range of PCR product size (bp):</td><td colspan="2" style="text-align: left;"><input style="text-align:center;" type="number" id="minpcr" min="30" max="999" value="150"/>-<input style="text-align:center;" type="number" id="maxpcr" min="50" max="5000" value="200"/></td></tr>
<tr><td>The gap between amplicons (bp):</td><td colspan="2" style="text-align: left;"><input style="text-align:center;" type="number" id="mingap" min="0" max="999" value="0"/></td></tr>
<tr><td colspan="3" style="text-align: left;"></td></tr>
<tr><td>Forward primer tail (5'-3'):</td><td colspan="2" style="text-align: left;"><input style="text-align:left;" type="text" name="ftail" id="ftail" size="45" placeholder="ACACTCTTTCCCTACACGACGCTCTTCCGATCT"/></td></tr>
<tr><td>Reverse primer tail (5'-3'):</td><td colspan="2" style="text-align: left;"><input style="text-align:left;" type="text" name="rtail" id="rtail" size="45" placeholder="GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT"/></td></tr>
<tr><td colspan="3" style="text-align: left;"></td></tr>
<tr><td colspan="2" style="text-align: left;"><button onclick="analyzeAndDisplay()" id="run_button" class="button button2">Generate</button></td><td><button onclick="document.getElementById('inputText').value= '';document.getElementById('analysisResult1').value = ''; document.getElementById('analysisResult2').value = '';" class="button button1">Clear</button></td></tr>
</table>

<span><strong>To export the results: select all (Ctrl-A), copy (Ctrl-C) and paste (Ctrl-V) to Excel sheet.</strong></span><br />
<div id="section2" class="section">
<div class="sub-tab-buttons"><button class="sub-tab-btn active" data-subtab="tab2-1">Primer list</button><button class="sub-tab-btn" data-subtab="tab2-2">PCR primer panels</button></div>
<div id="tab2-1" class="sub-tab-content" style="display: block;"><textarea id="analysisResult1" style="width:600px;height:300px"></textarea></div>
<div id="tab2-2" class="sub-tab-content"><textarea id="analysisResult2" style="width:600px;height:300px"></textarea></div>
</div><br /><br />

<li>Sequences are expected to be represented in the standard <a href="https://blast.ncbi.nlm.nih.gov/doc/blast-topics/" target="_blank">IUB/IUPAC nucleic acid codes</a> are acceptable letters: <b>B</b>=(C,G,T), <b>D</b>=(A,G,T), <b>H</b>=(A,C,T), <b>K</b>=(G,T), <b>M</b>=(A,C), <b> N</b>=(A,C,G,T), <b>R</b>=(A,G), <b>S</b>=(G,C), <b>V</b>=(A,C,G), <b>W</b>=(A,T), <b>Y</b>=(C,T), <b>U</b>=Uracil, <b>I</b>=Inosine.</li> 
<li><strong>Variants of the 3'-end composition (5'-3')</strong>: the structure of the last nucleotides at the 3'-end of the primer, it can be specified by "<span class="blue"><strong>N</strong></span>" for any pattern, or it can be encoded by one, two, three or more characters of standard or mixed letters. It is possible to specify one or more patterns (separated by spaces and of <strong>equal length</strong>): <strong><span class="blue">sws ssw sww wss www</strong></span>. For example, the pattern <span class="blue"><strong>WSS</strong></span> corresponds to all variants of the 3'-end composition: <span class="blue">acc acg agc agg tcc tcg tgc tgg</span>.</li>
<li><strong>Pre-designed list of primers/probes</strong>. Pre-designed primers/probes list is used for multiplexing with prior designed PCR primer/probe sets.</li> 
<li>Linguistic sequence complexity (LC%) is a measure of the 'vocabulary richness' of a genetic text based on counting the number of possible combinations of nucleotides ("entropy" of the set of possibilities) to the theoretically possible one. This value for sequence is converted to percentages, 100% being the highest level.</li> 
<li>Optionally, use two <b>‘<span class="red">/.../</span>’</b> signs for the start and end of the excluded region (this is possible multiple times).</li>
</ul><br />

<p>A single target sequence is allowed, either for the entire length or within the limiters (‘<b>[</b>‘ and ‘<b>]</b>’ ), or multiple target fragments within the sequence:<br />
<tt>
Single target: &nbsp;<span class="blue"> [<strong> Single target </strong>]</span><br />
Miltiple targets: &nbsp;<span class="blue"> [<strong> target 1 </strong>]</span>&nbsp;<span class="blue"> [<strong> target 2 </strong>]</span> &nbsp;<span class="blue"> [<strong> target 3 </strong>]</span> ....<br />
</tt></p>

<h3>&#1057; &gt;&gt; T bisulfite conversion (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3233226/" target="_blank">bisulfite modified genome</a>)</h3>
Sequence, design of specific PCR primers for <i>in silico</i> bisulphite conversion for both strands - only cytosines not followed by guanidine (CpG methylation) will be replaced by thymines:<br />
 <tt><br /> 5&rsquo;aa<span class="blue">CG</span>aagt<span class="blue">CCCCa</span>-3' &nbsp;&nbsp; &nbsp; &nbsp;&nbsp;5&rsquo;aa<span class="red">CG</span>aagt<span class="blue">TTTTa</span>-3'<br />
 &nbsp;&nbsp;||||||||||||| &nbsp; &nbsp; -&gt; &nbsp; &nbsp; &nbsp;||||||:|::::|</tt>&nbsp;<br />
<tt>3&rsquo;tt<span class="blue">GC</span>tt<span class="blue">C</span>aggggt<span class="blue"></span>-5' &nbsp;&nbsp; &nbsp;&nbsp; &nbsp;3&rsquo;tt<span class="red">GC</span>tt<span class="blue">T</span>aggggt</tt>
<br /><br />
<h3>Non-specific priming control</h3>
<p>Oligonucleotide specificity is one of the most critical factors for good PCR; optimal primers should hybridize only to the target sequence, especially when using complex genomic DNA as a template. Amplification problems can occur when primers anneal to repetitive sequences (retrotransposons, transposons or inverted tandem repeats). Alternative product amplification can also happen when primers are complementary to inverted repeats and produce multiple bands. However, the generation of inverted repeat sequences is exploited in two common generic DNA fingerprinting methods (RAPD).</p>
 

<p></p> 
 
</div>

<script src="../js/tabscript.js"></script>
<!--#include virtual="copyright.shtml" --> 
<!--#include virtual="footer.shtml" --> 
</body>
</html>