<!DOCTYPE html>
<html lang="en">
<head>
<title>PCR primer design tool</title>
<meta content="PCR, qPCR primer design tool" name="description" />
<!--#include virtual="header.shtml" --> 
<link rel="stylesheet" href="../css/soft2.css" type="text/css" />
<script src="../js/file-menu.js" type="text/javascript"></script>
<script src="../js/readseq.js" type="text/javascript"></script>
<script src="../js/pcr.js" type="text/javascript"></script>
<script src="../js/melting.js" type="text/javascript"></script>
<script src="../js/dna.js" type="text/javascript"></script>
<script src="../js/dimers.js" type="text/javascript"></script>
<script src="../js/displayseq.js" type="text/javascript"></script>
<script src="../js/ncbifasta.js" type="text/javascript"></script>
<script src="../js/ensembl.js" type="text/javascript"></script>
<script>
function openTab(evt, tabName) {
var i, tabcontent, tablinks;
tabcontent = document.getElementsByClassName("tab");
for (i = 0; i < tabcontent.length; i++) {
tabcontent[i].classList.remove("active");
}
tablinks = document.getElementsByClassName("tablink");
for (i = 0; i < tablinks.length; i++) {
tablinks[i].classList.remove("active");
}
document.getElementById(tabName).classList.add("active");
evt.currentTarget.classList.add("active");
}
</script>
<script>
function DisplaySeqs() {
const number_of_sequences = document.getElementById('number_of_sequences');
let result = Display(document.getElementById('inputText').value);
number_of_sequences.innerHTML = result;
}
</script>
<script>
function analyzeAndDisplay() {   
var word = analysis();
document.getElementById('analysisResult1').value = word[0];
document.getElementById('analysisResult2').value = word[1];
}
</script>
<script>
function handleClickTaqMan(checkbox){
const taqmancheckbox = document.getElementById('taqmanprobe');
if (taqmancheckbox.checked) {
const checkbox = document.getElementById('mgbprobe');
checkbox.checked = false;}
}
</script>
<script>
function handleClickMGB(checkbox){
const mgbcheckbox = document.getElementById('mgbprobe');
if (mgbcheckbox.checked) {
const checkbox = document.getElementById('taqmanprobe');
checkbox.checked = false;}  
}
</script>
</head>
<body id="contact" class="billboard">
<div id="page">
<!--#include virtual="menu.shtml" -->
<div class="span-13 home-coloum1"><br />
<div><a href=".">Back to Web Tools</a></div><br />

<h3>PCR, Multiplex, and QF-PCR primer design, genotyping tool</h3>
<p>This application offers advanced capabilities for designing primers across a wide range of PCR applications, including standard, inverse, multiplex, quantitative fluorescence (TaqMan and MGB-probe assay design), and bisulfite PCR. It also supports the development and validation of primer sets for genotyping single nucleotide polymorphisms (<strong>SNP</strong>) and insertions/deletions (<strong>InDel</strong>). Additionally, all individual tasks can be efficiently multiplexed for high-throughput analysis, such as fluorescence probe-based multiplex real-time qPCR assays.</p>
<p><strong>Input format:</strong> Sequences can be pasted or uploaded as a file in FASTA format or retrieved sequence (NCBI’s accession, e.g. A02710) from NCBI’s "nuccore" nucleotide database or retrieved flanked sequence <strong>SNP</strong> sequences from the Ensembl database. One or multiple SNP/variant <strong>rsIDs</strong> (comma or space separated: rs1357617 rs2046361 rs717302 rs1029047 rs917118) were used to retrieve the surrounding sequence (±flank bases) and enter the species name (<code>homo_sapiens</code>, <code>mus_musculus</code>, <code>rattus_norvegicus</code>, <code>bos_taurus</code>, <code>danio_rerio</code>, <code>arabidopsis_thaliana</code> etc).</p>
<p><strong>Size Limitations:</strong> the length of the query sequence and the size of the batch file are theoretically unlimited.</p>
<div>
<label for="speciesInput">Species:</label><input type="text" id="speciesInput" value="homo_sapiens" size="15" value="homo_sapiens" />
<label for="variantInput">rsIDs:</label><input type="text" id="variantInput" value="rs1357617 rs2046361 rs717302 rs1029047 rs917118" size="50" /><br />
<label for="flankInput">Flank size (&plusmn; bases):</label><input type="number" size="15" id="flankInput" value="300" min="100" />
<button onclick="fetchVariants()">Retrieve Sequences</button>
</div>
<br />
<div>
<label for="ncbiId">NCBI accession ID:</label><input type="text" id="ncbiId" id="ncbiId" value="A02710" />
<button onclick="fetchFastaSequence()">Retrieve Sequence</button>
</div><br />

<input type="file" id="fileInput">  
<strong><span class="blockHeader" id="number_of_sequences"></span></strong><br />
<div id="section1" class="section">
<strong>Input Format:</strong> Sequences can be pasted or uploaded as a file in FASTA format.<br />
<div class="sub-tab-buttons">
<button class="sub-tab-btn active" data-subtab="tab1-1">Target sequence(s)</button>
<button class="sub-tab-btn" data-subtab="tab1-2">Pre-designed list of primers/probes</button>
</div>
<div id="tab1-1" class="sub-tab-content" style="display: block;">
<textarea id="inputText" style="width:600px;height:250px" oninput="DisplaySeqs()">
>1 
[aggcagagagaatacggcttcaccgcaatatgaatcaggcgtagcagtaca]
tccgatggaagcctaaagttcacaaagtcagcttgccctagacacgttgtccagctgccgagcccccgttcatacagaaaacaagcaggattgtatgccacgctacacgtattcgacgaagccgcttttgcgcagagcgtgctcggctggcaagaattgatatacgca
[gctctcctaagcggggataccagagctccaacgtacgtagcagagggtgtcctttagtgcttggtctagaga]

>2 
[tatccacccagtgcacagcgtttaggtatttcatcacggctaatctagtttgacccaattcgacggtaacatcgtgccttcctgagtgtccaccattgttctagctccggaaagcccgtgatcgttccag]
ttcaacttcagtagtaagctcttatcatgcgcgatagcagtgcgccttgataaatcgtgcttgcaaacacataaccgttgagagtaaaattgaaat
[tgctcatgggcctggactcgctgtctaattcggttcatatatatgtagagaacaggtgactcgtttggggtcatgtcgagccgttaaatcaaagcacctgggccttaccttaactagacaagcctagcg]
</textarea>
</div>
<div id="tab1-2" class="sub-tab-content"><textarea id="inputPrimerList" style="width:600px;height:300px" placeholder=">M13-47&#13;cgccagggttttcccagtcacgac&#13;>RP&#13;tttcacacaggaaacagctatgac&#13;>RP-48&#13;agcggataacaatttcacacagga&#13;&#10;"></textarea></div>
</div><br />

<table border="0">
<tr><td colspan="2" style="text-align: left;"><span><strong>Primer design option</strong></span></td><td><strong>PCR applications</strong></td></tr>
<tr><td>Length range (nt):</td><td><input style="text-align:center;" type="number" id="minlen" min="12" max="99" value="20" />-<input style="text-align:center;" type="number" id="maxlen" min="12" max="99" value="21"/></td><td><input type="checkbox" id="multiplex" name="multiplex" value="multiplex"><label for="multiplex">Multiplex PCR</label></td></tr>
<tr><td>Tm range (°C):</td><td><input style="text-align:center;" type="number" id="mintm" min="37" max="80" value="61"/>-<input style="text-align:center;" type="number" id="maxtm" min="37" max="90" value="63"/></td><td><input type="checkbox" id="mgbprobe" name="mgbprobe" value="mgbprobe" onclick="handleClickMGB(this)"><label for="probe">MGB-probe design assay</label></td></tr>
<tr><td>Minimal linguistic complexity (%):</td><td><input style="text-align:center;" type="number" id="minlc" min="10" max="90" value="75"/></td><td><input type="checkbox" id="taqmanprobe" name="taqmanprobe" value="taqmanprobe" onclick="handleClickTaqMan(this)"><label for="taqmanprobe">TaqMan-probe design assay</label></td></tr>
<tr><td>Variants of the 3'-end composition (5'-3'):</td><td colspan="2" style="text-align: left;"><input style="text-align:left;" size="45" type="text" name="end3com" id="end3com" value="sws ssw sww wss www"/></td></tr>
<tr><td><input type="checkbox" id="repeats" name="repeats" value="repeats" checked ><label for="repeats">Non-specific priming control</label></td><td><input type="checkbox" id="overlapping" name="overlapping" value="overlapping" ><label for="overlapping">Overlaping primers</label></td><td><input type="checkbox" id="ctconvert" name="ctconvert" value="ctconvert"><label for="ctconvert">C>>T bisulfite conversion</label></td></tr>
<tr><td>Range of PCR product size (bp):</td><td><input style="text-align:center;" type="number" id="minpcr" min="30" max="999" value="200"/>-<input style="text-align:center;" type="number" id="maxpcr" min="50" max="5000" value="500"/></td><td><input type="checkbox" id="invertedPCR" name="invertedPCR" value="invertedPCR"><label for="invertedPCR">Inverse PCR (circular sequence)</label></td></tr>
<tr><td colspan="3" style="text-align: left;"></td></tr>
<tr><td>Forward primer tail (5'-3'):</td><td colspan="2" style="text-align: left;"><input style="text-align:left;" type="text" name="ftail" id="ftail" size="45" placeholder="tttcacacaggaaacagctatgac"/></td></tr>
<tr><td>Reverse primer tail (5'-3'):</td><td colspan="2" style="text-align: left;"><input style="text-align:left;" type="text" name="rtail" id="rtail" size="45" placeholder="agcggataacaatttcacacagga"/></td></tr>
<tr><td colspan="3" style="text-align: left;"></td></tr>
<tr><td colspan="2" style="text-align: left;"><button onclick="analyzeAndDisplay()" id="run_button" class="button button2">Generate</button></td><td><button onclick="document.getElementById('inputText').value= '';document.getElementById('analysisResult1').value = ''; document.getElementById('analysisResult2').value = '';" class="button button1">Clear</button></td></tr>
</table>

<span><strong>To export the results: select all (Ctrl-A), copy (Ctrl-C) and paste (Ctrl-V) to Excel sheet.</strong></span><br />
<div id="section2" class="section">
<div class="sub-tab-buttons"><button class="sub-tab-btn active" data-subtab="tab2-1">Primer list</button><button class="sub-tab-btn" data-subtab="tab2-2">PCR primer pairs</button></div>
<div id="tab2-1" class="sub-tab-content" style="display: block;"><textarea id="analysisResult1" style="width:600px;height:300px"></textarea></div>
<div id="tab2-2" class="sub-tab-content"><textarea id="analysisResult2" style="width:600px;height:300px"></textarea></div>
</div><br /><br />

<li>Sequences are expected to be represented in the standard <a href="https://blast.ncbi.nlm.nih.gov/doc/blast-topics/" target="_blank">IUB/IUPAC nucleic acid codes</a> are acceptable letters: <b>B</b>=(C,G,T), <b>D</b>=(A,G,T), <b>H</b>=(A,C,T), <b>K</b>=(G,T), <b>M</b>=(A,C), <b> N</b>=(A,C,G,T), <b>R</b>=(A,G), <b>S</b>=(G,C), <b>V</b>=(A,C,G), <b>W</b>=(A,T), <b>Y</b>=(C,T), <b>U</b>=Uracil, <b>I</b>=Inosine.</li> 
<li><strong>Variants of the 3'-end composition (5'-3')</strong>: the structure of the last nucleotides at the 3'-end of the primer, it can be specified by "<span class="blue"><strong>N</strong></span>" for any pattern, or it can be encoded by one, two, three or more characters of standard or mixed letters. It is possible to specify one or more patterns (separated by spaces and of <strong>equal length</strong>): <strong><span class="blue">sws ssw sww wss www</strong></span>. For example, the pattern <span class="blue"><strong>WSS</strong></span> corresponds to all variants of the 3'-end composition: <span class="blue">acc acg agc agg tcc tcg tgc tgg</span>.</li>
<li><strong>Pre-designed list of primers/probes</strong>. Pre-designed primers/probes list is used for multiplexing with prior designed PCR primer/probe sets.</li> 
<li>Linguistic sequence complexity (LC%) is a measure of the 'vocabulary richness' of a genetic text based on counting the number of possible combinations of nucleotides ("entropy" of the set of possibilities) to the theoretically possible one. This value for sequence is converted to percentages, 100% being the highest level.</li> 
<li>The user can specify individually for each sequence location for both <b><span class="blue">Forward</span></b> and <b><span class="red">Reverse</span></b> primers design using ‘<b>[</b>‘ and ‘<b>]</b>’ inside each sequence.</li>
<li>Optionally, use two <b>‘<span class="red">/.../</span>’</b> signs for the start and end of the excluded region (this is possible multiple times).</li>
</ul>

<p>All possible combinations of '<span class="blue">[&nbsp;Forward ]</span>' with '<span class="red">[&nbsp; Reverse ]</span>' within the sequence(s):<br />
<tt>
1.&nbsp;<span class="blue"> [</span><strong>SNP</strong>/<strong>InDel</strong><span class="red">]</span><br />
2.&nbsp;<span class="blue"> [&nbsp; ]</span>&nbsp;&nbsp;<span class="red">[</span>&nbsp;<span class="red"> ]</span><br />
3.&nbsp;<span class="blue"> [</span>&nbsp;<span class="red"> [</span>&nbsp;<span class="blue"> ]</span>&nbsp;<span class="red"> ]</span><br /> 
</tt></p>
<h3>&#1057; &gt;&gt; T bisulfite conversion (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3233226/" target="_blank">bisulfite modified genome</a>)</h3>
Sequence, design of specific PCR primers for <i>in silico</i> bisulphite conversion for both strands - only cytosines not followed by guanidine (CpG methylation) will be replaced by thymines:<br />
 <tt><br /> 5&rsquo;aa<span class="blue">CG</span>aagt<span class="blue">CCCCa</span>-3' &nbsp;&nbsp; &nbsp; &nbsp;&nbsp;5&rsquo;aa<span class="red">CG</span>aagt<span class="blue">TTTTa</span>-3'<br />
 &nbsp;&nbsp;||||||||||||| &nbsp; &nbsp; -&gt; &nbsp; &nbsp; &nbsp;||||||:|::::|</tt>&nbsp;<br />
<tt>3&rsquo;tt<span class="blue">GC</span>tt<span class="blue">C</span>aggggt<span class="blue"></span>-5' &nbsp;&nbsp; &nbsp;&nbsp; &nbsp;3&rsquo;tt<span class="red">GC</span>tt<span class="blue">T</span>aggggt</tt>
<br /><br />
<h3>Non-specific priming control</h3>
<p>Oligonucleotide specificity is one of the most critical factors for good PCR; optimal primers should hybridize only to the target sequence, especially when using complex genomic DNA as a template. Amplification problems can occur when primers anneal to repetitive sequences (retrotransposons, transposons or inverted tandem repeats). Alternative product amplification can also happen when primers are complementary to inverted repeats and produce multiple bands. However, the generation of inverted repeat sequences is exploited in two common generic DNA fingerprinting methods (RAPD).</p>
<h3><a href="https://www.biosearchtech.com/products/pcr-reagents-kits-and-instruments/pcr-probes-and-assays/custom-dna-probes-for-qpcr/minor-groove-binder-mgb-probes" target="_blank">Minor Groove Binders (MGBs)</a> Hydrolysis Probes</h3>
<p>Hydrolysis probes are typically labelled with a fluorophore reporter at the 5′end and a quencher [such as <a href="https://www.glenresearch.com/reports/gr33-11" target="_blank">the Eclipse Dark Quencher (EDQ)</a>] at the 3′end. During amplification, the DNA polymerase exonuclease activity cleaves off the reporter, allowing for signal unquenching and detection. Hydrolysis probes can be amended for increased binding stability or signal specificity. For example, MGBs selectively bind non-covalently to the minor groove, a shallow furrow in the DNA helix. Dual-labelled probes conjugated with MGB groups form extremely stable duplexes with single-stranded DNA targets, enabling shorter probe lengths and superior quenching.</p><br />

  
<br />
</div>

<script src="../js/tabscript.js"></script>
<!--#include virtual="copyright.shtml" --> 
<!--#include virtual="footer.shtml" --> 
</body>
</html>
